### PR TITLE
Stricter compile options

### PR DESCRIPTION
**Pull request by mck1117**
Turn back on a bunch of warnings-as-errors we had turned off, even though they never happen.

### CODE REVIEW

File: firmware/controllers/core/common_headers.h [hunk 1]
```diff
@@ -88,6 +88,8 @@
 #define EXPAND_Engine \
 	    engine_configuration_s *engineConfiguration = engine->engineConfigurationPtr; \
 		persistent_config_s *config = engine->config; \
+		(void)engineConfiguration; \
```
Comment (mck1117) [inline firmware/controllers/core/common_headers.h]: this saves tons of warnings about unused variable

**Comment by mck1117** [comment]
Uhh this had failing CI

**Comment by rusefi** [comment]
@mck1117 github CI is not really very reliable, was it a true failure?

**Comment by mck1117** [comment]
It should be a little more reliable now, and yes, this was a real failure

**Comment by rusefi** [comment]
Ooops :(

**Pull request merged**